pkg/predicate/predicate.go

```diff
@@ -350,3 +352,15 @@ func LabelSelectorPredicate(s metav1.LabelSelector) (Predicate, error) {
 		return selector.Matches(labels.Set(o.GetLabels()))
 	}), nil
 }
+
+// MatchingNamespacedNamePredicate constructs a Predicate to match a Namespaced object.
+// Only objects matching the Name and Namespace pair will be admitted.
+func MatchingNamespacedNamePredicate(namespacedName types.NamespacedName) (Predicate, error) {
+	if len(namespacedName.Name) == 0 {
+		return Funcs{}, fmt.Errorf("name cannot be empty")
+	}
+
+	return NewPredicateFuncs(func(object client.Object) bool {
```

Comment:
I am not sure I see much value in this Predicate over just defining the `NewPredicateFuncs` inline. And the fact that it can error makes it quite inconvenient to use